### PR TITLE
Lists and deletes leftover snapshots from CI jobs

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
go codebuild go

**Comment by lorengordon** [comment]
go codebuild go

**Comment by lorengordon** [comment]
This looks to be doing what we'd expect... deleting the snapshot is the new functionality...

```
Trying to deregister AMI: spel-ci-minimal-rhel-7-hvm-2019.08.dev1565722775.x86_64-gp2:ami-088aa4dccd0559dbf in us-east-1 
Trying to delete SNAPSHOT: snap-0e241054e8045b75c in us-east-1 
Trying to deregister AMI: spel-ci-minimal-centos-7-hvm-2019.08.dev1565722775.x86_64-gp2:ami-06a04788e1026b56e in us-east-1 
Trying to delete SNAPSHOT: snap-0f73688fbafe9c82b in us-east-1
```

File: build/post_build.sh [hunk 1]
```diff
@@ -3,12 +3,20 @@ echo "==========STARTING POST_BUILD=========="
...
-        # shellcheck disable=SC2030,SC2031,SC2091
-        $(export AWS_DEFAULT_REGION="$AWS_REGION"; test "$AMI_ID" != "None" && aws ec2 deregister-image --image-id "$AMI_ID" --profile "$SPEL_IDENTIFIER")
+        AMI_ID=$(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 describe-images --owners self --filters Name=name,Values="$AMI_NAME" --query 'Images[0].ImageId' --out text --profile "$SPEL_IDENTIFIER")
+
+        if [ "$AMI_ID" != "None" ]; then
+            # shellcheck disable=SC2030,SC2031
+            SNAPSHOT_ID=$(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 describe-images --image-id "$AMI_ID" --query Images[0].BlockDeviceMappings[0].Ebs.SnapshotId --out text --profile "$SPEL_IDENTIFIER")
+            echo "Trying to deregister AMI: ${AMI_NAME}:${AMI_ID} in ${AWS_REGION}"
+            # shellcheck disable=SC2030,SC2031,SC2091
+            $(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 deregister-image --image-id "$AMI_ID" --profile "$SPEL_IDENTIFIER")
```
Comment (ferricoxide) [inline build/post_build.sh]: I'm not understanding why you're prepending the subshell with a `$`: unless you're looking to collect the subshell's output into a variable, _should_ be able to invoke the subshell without it?

File: build/post_build.sh [hunk 2]
```diff
@@ -3,12 +3,20 @@ echo "==========STARTING POST_BUILD=========="
...
+            echo "Trying to delete SNAPSHOT: ${SNAPSHOT_ID} in ${AWS_REGION}"
+            # shellcheck disable=SC2030,SC2031,SC2091
+            $(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 delete-snapshot --snapshot-id "$SNAPSHOT_ID" --profile "$SPEL_IDENTIFIER")
```
Comment (ferricoxide) [inline build/post_build.sh]: Same as other comment/question.

**Comment by lorengordon** [comment]
go codebuild go

File: build/post_build.sh (hunk 1 above)
Comment (lorengordon) [reply build/post_build.sh]: fixed

**Pull request merged**